pkg/chunkenc/symbols.go

```diff
@@ -65,7 +70,15 @@ func (s *symbolizer) Add(lbls labels.Labels) symbols {
 	syms := make([]symbol, len(lbls))
 
 	for i, label := range lbls {
-		syms[i].Name = s.add(label.Name)
+		nameIdx := s.add(label.Name)
+		// Track that this symbol is a label name and store its normalized form
+		s.mtx.Lock()
+		if _, exists := s.normalizedNames[nameIdx]; !exists {
+			s.normalizedNames[nameIdx] = otlptranslator.NormalizeLabel(label.Name)
+		}
+		s.mtx.Unlock()
+
+		syms[i].Name = nameIdx
```

Comment:
Since we're doing it on lookup anyway, maybe we don't need to cache it in advance and it can always be lazily initialized?